### PR TITLE
Add 'starts with' to web custom format prefix.

### PR DESCRIPTION
**Pull request by snianu**
Closes https://github.com/w3c/clipboard-apis/pull/175#discussion_r1329392888

For normative changes, the following tasks have been completed:

 * [X] Modified Web platform tests (link to pull request): https://github.com/web-platform-tests/wpt/pull/41993

Implementation commitment:

 * [ ] WebKit (https://bugs.webkit.org/show_bug.cgi?id=)
 * [ ] Chromium (https://bugs.chromium.org/p/chromium/issues/detail?id=)
 * [ ] Gecko (https://bugzilla.mozilla.org/show_bug.cgi?id=)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/clipboard-apis/pull/195.html" title="Last updated on Oct 24, 2023, 8:29 PM UTC (e3ae9c5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/clipboard-apis/195/39b067c...e3ae9c5.html" title="Last updated on Oct 24, 2023, 8:29 PM UTC (e3ae9c5)">Diff</a>

### CODE REVIEW

**Review by sanketj — APPROVED** [review]
LGTM, thanks!

**Review by evanstade — COMMENTED** [review]
Sorry for delay --- I was OOO from the middle of last week.

File: index.bs [hunk 1]
```diff
@@ -717,7 +719,7 @@ url: https://w3c.github.io/permissions/#permissions-task-source; type: dfn;
 
 			1. Let |isCustom| be |false|.
 
-			1. If |type| has "web "("web" followed by U+0020 SPACE) prefix, then:
+			1. If |type| [=string/starts with=] "web "("web" followed by U+0020 SPACE) prefix, then:
```
[the remark this answers is not in the document]
Comment (snianu) [reply index.bs]: Done.

File: index.bs [hunk 2]
```diff
@@ -717,7 +719,7 @@ url: https://w3c.github.io/permissions/#permissions-task-source; type: dfn;
...
 			
 				1. Remove "web " prefix and set the remaining string to |type|.
```
[the remark this answers is not in the document]
Comment (snianu) [reply index.bs]: Done.

File: index.bs [hunk 3]
```diff
@@ -1695,7 +1697,7 @@ url: https://w3c.github.io/permissions/#permissions-task-source; type: dfn;
 
 				1. Let |webCustomFormat| be an empty {{Blob/type}}.
 
-				1. If |webCustomFormatString| has "web "("web" followed by U+0020 SPACE) prefix, then remove the "web " prefix and store the remaining string in |webCustomFormat|'s {{Blob/type}}'s [=MIME type/essence=], else abort all steps.
+				1. If |webCustomFormatString| [string/starts with=] "web "("web" followed by U+0020 SPACE) prefix, then remove the "web " prefix and store the remaining string in |webCustomFormat|'s {{Blob/type}}'s [=MIME type/essence=], else abort all steps.
```
[the remark this answers is not in the document]
Comment (snianu) [reply index.bs]: Added MIME type parsing checks. Thanks!

**Comment by snianu** [comment]
@evanstade Addressed your comments. PTAL.

File: index.bs [hunk 4]
```diff
@@ -539,9 +539,8 @@ url: https://w3c.github.io/permissions/#permissions-task-source; type: dfn;
 
 			* text/uri-list
 			* image/svg+xml
-			* Custom format [=string/starts with=] "web "("web" followed by U+0020 SPACE) prefix
-				and suffix (after stripping out "web "("web" followed by U+0020 SPACE)) is a valid
-				[=/MIME type=].
+			* Custom format [=string/starts with=] `"web "`("web" followed by U+0020 SPACE) prefix
+				and suffix (after stripping out `"web "`) is a valid [=/MIME type=].
```
Comment (evanstade) [inline index.bs:543]: in the MIME sniffing standard I found:

> "text/html;" is not a [valid MIME type string](https://mimesniff.spec.whatwg.org/#valid-mime-type), though [parse a MIME type](https://mimesniff.spec.whatwg.org/#parse-a-mime-type) returns a [MIME type record](https://mimesniff.spec.whatwg.org/#mime-type) for it identical to if the input had been "text/html".

So do we really want to require this to be a valid mime type? The "new ClipboardItem" steps, as well as the algorithm that's touched below, we don't require validity, just that parsing succeeds.

File: index.bs [hunk 5]
```diff
@@ -671,9 +670,9 @@ url: https://w3c.github.io/permissions/#permissions-task-source; type: dfn;
 
 				1. Let |isCustom| be |false|.
 				
-				1. If |key| [=string/starts with=] "web " ("web" followed by U+0020 SPACE) prefix, then
+				1. If |key| [=string/starts with=] `"web "` prefix, then
 				
-					1. Remove "web " prefix and set the remaining string to |key|.
+					1. Remove `"web "` prefix and set the remaining string to |key|.
```
Comment (evanstade) [inline index.bs:675]: nit: `assign the remaining string to |key|` or `set |key| to the remaining string`

**Review by evanstade — CHANGES_REQUESTED** [review]

File: index.bs (hunk 5 above)
Comment (snianu) [reply index.bs:675]: Done.

**Comment by snianu** [comment]
@evanstade Addressed your comments. PTAL.

File: index.bs [hunk 6]
```diff
@@ -717,9 +718,9 @@ url: https://w3c.github.io/permissions/#permissions-task-source; type: dfn;
 
 			1. Let |isCustom| be |false|.
 
-			1. If |type| has "web "("web" followed by U+0020 SPACE) prefix, then:
+			1. If |type| [=string/starts with=] `"web "` prefix, then:
 			
-				1. Remove "web " prefix and set the remaining string to |type|.
+				1. Remove `"web "` prefix and set the remaining string to |type|.
```
Comment (evanstade) [inline index.bs:723]: nit: here too

**Review by evanstade — CHANGES_REQUESTED** [review]
lgtm

**Pull request merged**